src/Dropdown/Dropdown.tsx

```diff
@@ -1,400 +1,160 @@
-  };
-
-  componentDidMount() {
-    this.logWarnings();
-  }
-
-  logWarnings = () => {
-    if (this.props.children) {
-      warn("You're passing children. Not expected behaviour");
-    }
-  };
-
-  valueToOption = (
-    value: Dropdown.Value | undefined,
-    options: SelectNS.Options
-  ) => {
-    if (t.String.is(value) || t.Number.is(value)) {
-      const { multi, delimiter } = this.props as DefaultedProps;
-      if (multi) {
-        const values = String(value).split(delimiter);
-        return compact(
-          values.map(v => find(options, { value: v }) || undefined)
-        );
-      }
-      return find(options, { value }) || undefined;
-    }
-    return value;
-  };
-
-  sortOptionsByGroup = (options: SelectNS.Options) => {
-    const { groupByKey } = this.props as DefaultedProps;
-    return sortBy(
-      options,
-      option =>
-        option[groupByKey]
-          ? findIndex(options, o => option[groupByKey] === o[groupByKey])
-          : -1
-    );
+    blurInputOnSelect: true,
+    menuPosition: "bottom",
+    components: {}
   };
 
   getCustomClassNames() {
-    const { size, flat, clearable, menuPosition } = this
-      .props as DefaultedProps;
+    const {
+      size,
+      flat,
+      isClearable,
+      menuPosition,
+      isMulti,
+      isDisabled
+    } = this.props;
     return cx({
       "is-medium": size === "medium",
       "is-small": size === "small",
       "is-flat": flat,
-      "is-clearable": clearable,
+      "is-multi": isMulti,
+      "is-clearable": isClearable,
+      "is-disabled": isDisabled,
       "menu-position-top": menuPosition === "top"
     });
   }
 
-  _onChange = (_value?: Dropdown.Value | null) => {
-    const value = _value === "" || isEmptyArray(_value) ? null : _value;
-    return this.props.onChange(value);
-  };
-
-  onInputKeyDown: SelectNS.OnInputKeyDownHandler = e => {
-    if (e.keyCode === 38 || e.keyCode === 40) {
-      if (isEmptyArray(this.props.options)) {
-        e.preventDefault();
-      }
-    }
-  };
+  private select: Select<OptionType> | null;
 
-  private select: Select | null;
   focus = () => {
     this.select && this.select.focus();
   };
 
-  optionGroupRenderer = (title?: string) => {
-    return title ? (
-      <FlexView className="Select-option-group">
-        {(this.props as DefaultedProps).optionGroupRenderer(title)}
-      </FlexView>
-    ) : null;
-  };
-
-  menuRenderer = (
-    args: SelectNS.MenuRendererProps & Dropdown.SelectMissingMenuRendererProps
-  ) => {
-    const { menuRenderer, groupByKey } = this.props as DefaultedProps;
-    return menuRenderer({
-      ...args,
-      groupByKey,
-      optionGroupRenderer: this.optionGroupRenderer
-    });
-  };
-
   render() {
     const {
-      _onChange,
-      onInputKeyDown,
-      props: { className, options, backspaceRemoves, clearable, ...props }
+      props: {
+        className,
+        backspaceRemovesValue,
+        isClearable,
+        menuPosition,
```

Comment:
`menuPosition` is not used => never passed to react-select through `...props`

Reply:
`menuPosition` is defined here with the explicit intention of not passing it to `react-select` as it is now an accepted prop but we want to keep our implementation (or maybe not? what do you think?)

Reply:
ok, now it's clear 👍 

> (or maybe not? what do you think?)
what are the benefits of our own implementation compared to react-select one?